http://pad.lv/1715738
https://bugs.launchpad.net/ubuntu/+source/cloud-init/+bug/1715738

=== Begin SRU Template ===
[Impact]
Skipping modules based on invalid distro setting was not honored, so invalid modules could run/fail.

[Test Case]
# create a cloud-config with invalid module config for ubuntu
cat >run-unverified-spacewalk.yaml <<EOF
#cloud-config
unverified_modules: [spacewalk]
EOF

## launch an instance enabling spacewalk in cloud.cfg and validate logs
for release in xenial zesty; do
  ref=$release-proposed;
  name=test-$release
  lxc-proposed-snapshot --proposed --publish $release $ref;
  lxc init $ref $name;
  lxc init $ref $name-2;
  # Add ubuntu-invalid spacewalk module to list of cloud config modules
  sudo sed -i  's/ - byobu/ - byobu\n - spacewalk\n/' /var/lib/lxd/containers/$name/rootfs/etc/cloud/cloud.cfg;
  sudo sed -i  's/ - byobu/ - byobu\n - spacewalk\n/' /var/lib/lxd/containers/$name-2/rootfs/etc/cloud/cloud.cfg;
  lxc config set $name-2 user.user-data - < run-unverified-spacewalk.yaml
  lxc start $name;
  lxc start $name-2;
  sleep 10;
  echo checking $name;
  lxc exec $name -- grep 'space' # expect to see a Skip log and no 'Running config-spacewalk'
  echo checking $name-2;
  lxc exec $name-2 -- grep 'space' # expect no Skip log and presence of 'Running config-spacewalk'
done


[Regression Potential]
Minimal just log timestamp formatting changes

[Other Info]
Upstream commit at
  https://git.launchpad.net/cloud-init/commit/?id=f761f2b5

=== End SRU Template ===

=== SRU verification output ===
Creating xenial-proposed-1302819984
--proposed -- 
Get:1 http://security.ubuntu.com/ubuntu xenial-security InRelease [102 kB]
...
Get:1 http://archive.ubuntu.com/ubuntu xenial-proposed/main amd64 cloud-init all 17.1-18-gd4f70470-0ubuntu1~16.04.1 [342 kB]
Preconfiguring packages ...
Creating test-xenial
Creating test-xenial-2
checking test-xenial
2017-10-17 16:33:21,666 - stages.py[INFO]: Skipping modules 'spacewalk' because they are not verified on distro 'ubuntu'.  To run anyway, add them to 'unverified_modules' in config.
checking test-xenial-2
2017-10-17 16:33:22,038 - stages.py[INFO]: running unverified_modules: 'spacewalk'
2017-10-17 16:33:22,880 - stages.py[DEBUG]: Running module spacewalk (<module 'cloudinit.config.cc_spacewalk' from '/usr/lib/python3/dist-packages/cloudinit/config/cc_spacewalk.py'>) with frequency once-per-instance
2017-10-17 16:33:22,880 - handlers.py[DEBUG]: start: modules-config/config-spacewalk: running config-spacewalk with frequency once-per-instance
2017-10-17 16:33:22,881 - util.py[DEBUG]: Writing to /var/lib/cloud/instances/test-xenial-2/sem/config_spacewalk - wb: [644] 24 bytes
2017-10-17 16:33:22,881 - helpers.py[DEBUG]: Running config-spacewalk using lock (<FileLock using file '/var/lib/cloud/instances/test-xenial-2/sem/config_spacewalk'>)
2017-10-17 16:33:22,881 - cc_spacewalk.py[DEBUG]: Skipping module named spacewalk, no 'spacewalk' key in configuration
2017-10-17 16:33:22,881 - handlers.py[DEBUG]: finish: modules-config/config-spacewalk: SUCCESS: config-spacewalk ran successfully


Creating zesty-proposed-244714121
--proposed -- 
Hit:1 http://archive.ubuntu.com/ubuntu zesty InRelease
...
Get:1 http://archive.ubuntu.com/ubuntu zesty-proposed/main amd64 cloud-init all 17.1-18-gd4f70470-0ubuntu1~17.04.1 [340 kB]
Preconfiguring packages ...
Creating test-zesty
Creating test-zesty-2
checking test-zesty
2017-10-17 16:34:43,271 - stages.py[INFO]: Skipping modules 'spacewalk' because they are not verified on distro 'ubuntu'.  To run anyway, add them to 'unverified_modules' in config.
checking test-zesty-2
2017-10-17 16:34:43,263 - stages.py[INFO]: running unverified_modules: 'spacewalk'
2017-10-17 16:34:44,097 - stages.py[DEBUG]: Running module spacewalk (<module 'cloudinit.config.cc_spacewalk' from '/usr/lib/python3/dist-packages/cloudinit/config/cc_spacewalk.py'>) with frequency once-per-instance
2017-10-17 16:34:44,097 - handlers.py[DEBUG]: start: modules-config/config-spacewalk: running config-spacewalk with frequency once-per-instance
2017-10-17 16:34:44,097 - util.py[DEBUG]: Writing to /var/lib/cloud/instances/test-zesty-2/sem/config_spacewalk - wb: [644] 23 bytes
2017-10-17 16:34:44,097 - helpers.py[DEBUG]: Running config-spacewalk using lock (<FileLock using file '/var/lib/cloud/instances/test-zesty-2/sem/config_spacewalk'>)
2017-10-17 16:34:44,097 - cc_spacewalk.py[DEBUG]: Skipping module named spacewalk, no 'spacewalk' key in configuration
2017-10-17 16:34:44,097 - handlers.py[DEBUG]: finish: modules-config/config-spacewalk: SUCCESS: config-spacewalk ran successfully

=== SRU verification output ===

